packages/twenty-front/src/modules/workflow/hooks/useWorkflowWithCurrentVersion.tsx

```diff
@@ -22,34 +23,37 @@ export const useWorkflowWithCurrentVersion = (
     skip: !isDefined(workflowId),
   });
 
-  const { records: mostRecentWorkflowVersions } =
-    useFindManyRecords<WorkflowVersion>({
-      objectNameSingular: CoreObjectNameSingular.WorkflowVersion,
-      filter: {
-        workflowId: {
-          eq: workflowId,
-        },
+  const {
+    records: mostRecentWorkflowVersions,
+    loading: loadingMostRecentWorkflowVersions,
+  } = useFindManyRecords<WorkflowVersion>({
+    objectNameSingular: CoreObjectNameSingular.WorkflowVersion,
+    filter: {
+      workflowId: {
+        eq: workflowId,
       },
-      orderBy: [
-        {
-          createdAt: 'DescNullsLast',
-        },
-      ],
-      limit: 1,
-      skip: !isDefined(workflowId),
-    });
+    },
+    orderBy: [
+      {
+        createdAt: 'DescNullsLast',
+      },
+    ],
+    limit: 1,
+    skip: !isDefined(workflowId),
+  });
+
+  const workflowWithCurrentVersion = useMemo(() => {
```

Comment:
I memoize because we expect a stable reference to be returned by this function. We expect the returned workflow data to change to only when the workflow really changes and not because the hook was re-rendered.

Reply:
not sure it's super useful here. Why do we need a stable reference?

Reply:
It was helpful when I relied on the `workflowWithCurrentVersion` in a `useEffect` to create the first workflow version conditionally.
It makes sense for a hook to return the same object reference until the object changes. That's how the `useQuery` hook of Apollo React works by caching data instead of returning a new reference at each render.

However, strictly speaking, it's not necessary as we don't use a useEffect depending on that variable anymore.

Reply:
UPDATE: I'm still using a useEffect in the WorkflowEffect component. I find it more predictable to memoize `workflowWithCurrentVersion` as the useEffect depends on its `currentVersion` property.

Reply:
ok :) I'm not against memoization when it's not hiding a deeper issue (like API calls in useEffects) ! LGTM